### ISSUE TITLE
Add Option for Automatic Exception on Empty EntitySelectorArgument Lists

### ISSUE THREAD

**Issue by learliet** [JorelAli/CommandAPI#495]
### Description

There should be an option available when utilizing the `EntitySelectorArgument` to trigger an exception automatically when the resulting list is empty. This behavior mirrors the default implementation where _argument.entity.notfound.entity_ or _argument.entity.notfound.player_ is respectively returned.

### Expected code

```kotlin
val myCommand = CommandAPICommand("myCommand")
  .withArguments(EntitySelectorArgument.ManyPlayers("players", false)) // allowEmpty flag
  .executes(CommandExecutor { sender, args ->
      // automatically returns "argument.entity.notfound.player" if empty
      val players = args[0] as List<Player> 
      // [...]
  })
```

**Comment by JorelAli** [comment]
I like this idea, and it should be fairly trivial to implement. In our existing multiple entity selector code, we have this:

https://github.com/JorelAli/CommandAPI/blob/38a1a7682633a2095366095db183850e1a9d99eb/commandapi-platforms/commandapi-bukkit/commandapi-bukkit-nms/commandapi-bukkit-1.20/src/main/java/dev/jorel/commandapi/nms/NMS_1_20_R1.java#L353-L372

In this implementation, we catch the exception when the list of entities would be empty and instead return an empty array. Adding a parameter to allow this exception to bubble up should be available as an option.

It would be nice to get this implemented for our next release given how simple this is to implement.

**Comment by JorelAli** [comment]
Implemented in version 9.4.0.
